hello.py

```diff
@@ -111,12 +111,12 @@ def syst_name_in_url(e):
 def load_name(filename):
     if filename!=filename.upper():
         abort(406, filename)
-    if dp.is_systematic_name(filename) and not dp.is_hypothetical_protein(filename) :
+    if np.is_systematic_name(filename) and not np.is_hypothetical_protein(filename) :
         abort(410, filename)
 
-    filepath = os.path.join(os.path.expanduser('~/data'),dp.systematic_name(filename)+'.0.ssw11.hhr')
+    filepath = os.path.join(os.path.expanduser('~/data'),np.systematic_name(filename)+'.0.ssw11.hhr')
     # TODO: When the reference file is complete, remove the following
-    if dp.is_unknown_protein(filename):
+    if np.is_unknown_protein(filename):
```

Comment:
Hmm, except here unknown & hypothetical proteins are handled differently :(

I don't understand the logic of what the different kinds of names are to figure out what the correct logic should be!

Reply:
A hypothetical protein is one with a legitimate systematic name, but no standard name, so it exists in one of the maps. An unknown protein is one that doesn't exist in any of the maps - it could be a legitimate one but our maps are incomplete, or a totally bonkers one that a user's cat typed on the url. 

I think the new refactoring sheds some light into this, but I'll add a few more comments anyways.